integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiJmsRecovery.java

```diff
@@ -0,0 +1,516 @@
+import static oracle.weblogic.kubernetes.actions.ActionConstants.WORK_DIR;
+import static oracle.weblogic.kubernetes.actions.TestActions.createDomainCustomResource;
+import static oracle.weblogic.kubernetes.actions.TestActions.createSecret;
+import static oracle.weblogic.kubernetes.actions.TestActions.execCommand;
+import static oracle.weblogic.kubernetes.actions.TestActions.getServiceNodePort;
+import static oracle.weblogic.kubernetes.actions.TestActions.listServices;
+import static oracle.weblogic.kubernetes.actions.TestActions.scaleCluster;
+import static oracle.weblogic.kubernetes.assertions.TestAssertions.domainExists;
+import static oracle.weblogic.kubernetes.utils.CommonMiiTestUtils.createDomainSecret;
+import static oracle.weblogic.kubernetes.utils.CommonMiiTestUtils.createJobToChangePermissionsOnPvHostPath;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.checkPodDoesNotExist;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.checkPodReadyAndServiceExists;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.createConfigMapAndVerify;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.createOcirRepoSecret;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.createPV;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.createPVC;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.createSecretForBaseImages;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.getExternalServicePodName;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.installAndVerifyOperator;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.runClientInsidePod;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.runJavacInsidePod;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.setPodAntiAffinity;
+import static oracle.weblogic.kubernetes.utils.DbUtils.startOracleDB;
+import static oracle.weblogic.kubernetes.utils.ExecCommand.exec;
+import static oracle.weblogic.kubernetes.utils.FileUtils.copyFileToPod;
+import static oracle.weblogic.kubernetes.utils.ThreadSafeLogger.getLogger;
+import static org.awaitility.Awaitility.with;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * This test class verifies JMS Service migration when a cluster is scaled 
+ * down by removing the pod. The usecase verifies the JMS service on shutdown 
+ * pod/server is migrated to one of the active server and the JMS messages
+ * can be recovered after service migration.
+ * The dynamic cluster is configured with an ORACLE leasing datasource.
+ * The cluster targeted persistent store is configured with 'Dynamic' 
+ * distribution policy and 'Always' migration policy.
+ * The associated file store directory is on a shared volume so that it can 
```

Comment:
From the description its not clear the test class verifies JMS service migration from 2 different persistent stores separately. Can you mention that it tests both stores and it is independent of each other.

Reply:
/**
 * This test class verifies JMS Service migration with JMS messages stored in 
 * (a) Persistent FileStore (b) Persistent JDBC Store accessible from all pods.
 * The dynamic cluster is scaled down to trigger the Service migration from 
 * a stopped pod/managed server to a live pod/managed server.
 * Configuration : 
 *   MII cluster domain with 2 managed servers
 *   Two set of JMS Resources with FileStore and JDBC Store
 *   All resources are targeted to cluster with enabled JMS service migration
 *   Two Distributed Queue(s) one with FileStore and the other with JDBC Store
 *   Separate ORACLE Datasource for cluster leasing
 * UseCase :  
 * (a) Test client sends 100 messages to member queue@managed-server2
 * (b) Scale down the cluster with replica count 1 to shutdown managed-server2
 * (c) Make sure the JMS Service@managed-server2 is migrated to managed-server1
 * (d) Make sure all 100 messages got recovered once the 
 *     JMS Service@managed-server2 is migrated to managed-server1 
 * */